graylog2-server/src/main/java/org/graylog2/alarmcallbacks/EmailAlarmCallback.java

```diff
@@ -106,6 +95,20 @@ public void call(Stream stream, AlertCondition.CheckResult result) {
         }
     }
 
+    protected List<Message> getAlarmBacklog(AlertCondition.CheckResult result) {
+        final AlertCondition alertCondition = result.getTriggeredCondition();
+        final int effectiveBacklogSize = Math.min(alertCondition.getBacklog(), result.getMatchingMessages().size());
+        final List<MessageSummary> backlogSummaries = result.getMatchingMessages()
```

Comment:
I'd rather do a short circuit here and return an empty list if `effectiveBacklogSize` is 0.